python/examples/advanced_agents/assignment-agent/assignment_agent.py

```diff
@@ -0,0 +1,167 @@
+from autogen import AssistantAgent, UserProxyAgent
+from composio_autogen import ComposioToolSet, Action
+from composio import AppType
+import os
+from dotenv import load_dotenv
+
+load_dotenv()
+
+toolset = ComposioToolSet(
+    api_key=os.getenv("COMPOSIO_API_KEY"),
+    entity_id="karthik",
+)
+llm_config = {
+    "config_list": [
+        {
+            "model": "gpt-4o",
+            "api_key": os.getenv("OPENAI_API_KEY"),
+        }
+    ]
+}
+
+chatbot = AssistantAgent(
+    "chatbot",
+    system_message="You're a helpful assistant that creates assignments for a course & reviews them. You will be given the details of the assigment to create and review, after creating or reviewing you need to TERMINATE.",
+    llm_config=llm_config,
+)
+
+user_proxy = UserProxyAgent(
+    name="User",
+    is_termination_msg=lambda x: x.get("content", "")
+    and "TERMINATE" in x.get("content", ""),
+    human_input_mode="NEVER",
+    code_execution_config={"use_docker": False},
+)
+
+toolset.register_tools(
+    actions=[
+        Action.CANVAS_CREATE_ASSIGNMENT,
+        Action.CANVAS_GET_ASSIGNMENT,
+        Action.CANVAS_LIST_ASSIGNMENT_SUBMISSIONS,
+        Action.CANVAS_GRADE_COMMENT_SUBMISSION,
+    ],
+    caller=chatbot,
+    executor=user_proxy,
+)
+
+def get_courses():
+    courses = toolset.execute_action(
```

Comment:
Consider adding error handling for API calls. The function should catch potential exceptions that might occur during the API call and handle them gracefully. For example:
```python
def get_courses():
    try:
        courses = toolset.execute_action(
            action=Action.CANVAS_LIST_COURSES,
            entity_id="karthik",
            params={}
        )
        return courses
    except Exception as e:
        print(f"Error fetching courses: {str(e)}")
        return None
```